presto-main/src/main/java/io/prestosql/sql/planner/LogicalPlanner.java

```diff
@@ -182,7 +188,12 @@ public Plan plan(Analysis analysis, Stage stage)
         TypeProvider types = symbolAllocator.getTypes();
         StatsProvider statsProvider = new CachingStatsProvider(statsCalculator, session, types);
         CostProvider costProvider = new CachingCostProvider(costCalculator, statsProvider, Optional.empty(), session, types);
-        return new Plan(root, types, StatsAndCosts.create(root, statsProvider, costProvider));
```

Comment:
While I think the change is valueable and should be ON by default, it's also backward incompatible for people having `EventListener`s. Make sure we communicate it clearly.

Reply:
i guess, i just mean to call it out there